xl-v1-createfile.go

```diff
@@ -51,31 +51,60 @@ func closeAndRemoveWriters(writers ...io.WriteCloser) {
 // WriteErasure reads predefined blocks, encodes them and writes to
 // configured storage disks.
 func (xl XL) writeErasure(volume, path string, reader *io.PipeReader) {
-	var writers = make([]io.WriteCloser, len(xl.storageDisks))
-	var sha512Writers = make([]hash.Hash, len(xl.storageDisks))
-	var metadataWriters = make([]io.WriteCloser, len(xl.storageDisks))
-
```

Comment:
As per my reading, `append()` doesn't change order in returning slice and if we need to have very controlled elements count for disks, we would need to return/accept slices for disks in functions. similarly `XL.storageDisks` has to be immutable variable.

Anyways I am fixing by having array.